circuitv2: clarify what to do when data limit is reached (#526)
libp2p/specs: relay/circuit-v2.md

accidental termination according to its connection management policy.
If a relay server becomes overloaded however, it may still drop a
connection with reservations in order to maintain its resource quotas.
  

> If more data than the limit specified in the `data` field is transferred
> over the connection, the server should reset the stream. If the reservation

Reviewer: ```suggestion
If more data than the limit specified in the `data` field is transferred
over the relayed connection, or the relayed connection has been open for longer than `duration`, the relay should reset the stream to the source and the stream to the destination. If the reservation
```

Author: Much better, thanks!

> for the connection has expired the server may then apply any connection

Reviewer 2: Who's the "server" here. Is it the relay? We probably shouldn't be using 2 different terms here.

The following sentence doesn't really make sense to me, as a relayed connection doesn't really show up as a normal connection in the connection manager.

Reviewer 3: > as a relayed connection doesn't really show up as a normal connection in the connection manager.

That sounds a bit too implementation-specific to me? I do agree though that the sentence is superfluous. "Do what you'd normally do" isn't really worth "specifying" I'd say.

management policy to the connection as normal otherwise it should retain

> the connection, unless doing so would prevent it from mainting it's resource

Reviewer 3: ```suggestion
the connection, unless doing so would prevent it from maintining it's resource
```

Author: ```suggestion
the connection, unless doing so would prevent it from maintaining it's resource
```

Spelling is hard 🤣

Reviewer 2: And there's another mistake ;)

Author: ```suggestion
the connection, unless doing so would prevent it from maintaining its resource
```

Arrgh!

> the connection, unless doing so would prevent it from maintaining it's resource

Author: ```suggestion
the connection, unless doing so would prevent it from maintaining its resource
```
